ts/features/bonus/bpd/store/reducers/details/selectedPeriod.ts

```diff
@@ -0,0 +1,38 @@
+import { NavigationActions } from "react-navigation";
+import { createSelector } from "reselect";
+import { getType } from "typesafe-actions";
+import { Action } from "../../../../../../store/actions/types";
+import { GlobalState } from "../../../../../../store/reducers/types";
+import BPD_ROUTES from "../../../navigation/routes";
+import { BpdPeriod } from "../../actions/periods";
+import { bpdSelectPeriod } from "../../actions/selectedPeriod";
+
+/**
+ * Store the current period selected by the user (current displayed)
+ * @param state
+ * @param action
+ */
+export const bpdSelectedPeriodsReducer = (
+  state: BpdPeriod | null = null,
+  action: Action
+): BpdPeriod | null => {
+  switch (action.type) {
+    case getType(bpdSelectPeriod):
+      return action.payload;
+    case NavigationActions.NAVIGATE:
+      if (action.routeName === BPD_ROUTES.DETAILS) {
```

Comment:
I guess here we store some fields curried by the navigation.
Is it right?
Could you leave a comment explaining?

Reply:
I've added a comment in eaef183! Yes, we use the optional payload of the navigation action to set the right `selectedPeriod` (the user navigate to a specific detail period).